#! /bin/sh /usr/share/dpatch/dpatch-run
## 36_fix_erf_crash.dpatch by  <balint@balintreczey.hu>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix erf crash, based on fix released in Wireshark 1.2.2

@DPATCH@

--- trunk/wiretap/erf.c.orig	2009-11-17 00:31:05.000000000 +0100
+++ trunk/wiretap/erf.c	2009-11-17 00:39:22.000000000 +0100
@@ -404,6 +404,16 @@
 		*packet_size -= skip;
 	}
 
+	if (*packet_size > WTAP_MAX_PACKET_SIZE) {
+		/*
+		 * Probably a corrupt capture file; don't blow up trying
+		 * to allocate space for an immensely-large packet.
+		 */
+		*err = WTAP_ERR_BAD_RECORD;
+		*err_info = g_strdup_printf("erf: File has %u-byte packet, bigger than maximum of %u",
+		    *packet_size, WTAP_MAX_PACKET_SIZE);
+		return FALSE;
+	}
 	return TRUE;
 }
 
